source/js/components/join/join.jsx

```diff
@@ -224,28 +235,39 @@ export default class JoinUs extends React.Component {
    * Render the email field in signup CTA.
    */
   renderEmailField() {
-    let classes = classNames(`mb-2`, {
+    let wrapperClasses = classNames({
       "has-danger":
         (!this.state.apiSuccess &&
           this.state.userTriedSubmitting &&
-          !this.validatesAsEmail(this.email.value)) ||
+          this.validatesAsEmail(this.email.value).errorMessage) ||
```

Comment:
I tried storing this as a variable and each time, I get the error that `this.email` is undefined for some odd reason.

Reply:
Ah I see. This is because we are using `ref`. We should eventually rewrite this so we use `this.state` to store form values instead of binding fields using `ref` and access value using this.refName.value. However, this is out of the scope of this ticket.

Let's add this to `renderEmailField() {`,

```jsx
renderEmailField() {
  // During the first render, we bind the email field to this.email
  // using ref={el => (this.email = el)}
  // This means this.email is undefined until the second render.
  // To avoid TypeError we have do the following conditonal assignment.
  let emailValidation = this.email
    ? this.validatesAsEmail(this.email.value)
    : false;
```

(feel free to edit the inline comments so the wording sounds better for you)

Reply:
To check if email value is valid let's do it more explicitly (e.g., Instead of checking `errorMessage` we check the `valid` property.)

So we do `emailValidation.valid` to check if email is valid. If email is invalid, we then render the error message using `emailValidation.errorMessage`.